src/pages/Chat/ChatConversation.tsx

```diff
@@ -1,35 +1,102 @@
 import { ChatMessage } from 'components/hedvig/chat'
 import { Conversation, Message } from 'components/hedvig/conversation'
+import { Container } from 'constate'
+import { allPass, isNil, not, path, pipe, prop } from 'ramda'
 import * as React from 'react'
+import { Mount } from 'react-lifecycle-components'
+import { ChatContainer, State as ChatState } from './state'
 import { Greet } from './steps/Greet'
 import { NameAgeInput } from './steps/NameAgeInput'
 
+interface InitialStepState {
+  hasSetInitialStep: boolean
+}
+
+interface InitialStepActions {
+  setInitialStep: () => void
+}
+
+const step1PropIsntNil = (step1Prop: string) => (steps: ChatState) =>
+  pipe(
+    path(['step1', step1Prop]),
+    isNil,
+    not,
+  )(steps)
+
+const getInitialStep = (steps: ChatState) => {
+  if (
+    allPass([
+      pipe(
+        prop('step1'),
+        isNil,
+        not,
+      ),
+      step1PropIsntNil('firstName'),
+      step1PropIsntNil('lastName'),
+      step1PropIsntNil('age'),
+    ])(steps)
+  ) {
+    return 5
```

Comment:
what does this magic number mean? ✨

Reply:
hehe, idk how to do it more neat currently, but it's the index of the message that we should just `appear` ( = this is where they are in the flow). May need to refactor this into a more clever solution when we get a more complex flow with branches n stuff. But is it OK for now or not really?